### ISSUE TITLE
isinstance doesn't work with interfaces

### ISSUE THREAD

**Issue by tonyroberts** [pythonnet/pythonnet#51]
eg

```python
from System import String, IComparable
s = String("")

# doesn't work
isinstance(s, IComparable)
>> False

# works
s.GetType().GetInterface("IComparable")
>> <System.RuntimeType at 0xaccb048>
```

I think it just needs tp_bases to be set in the typemanager instead of tp_base.

**Comment by filmor** [comment]
Could you give more details on how this could be implemented?

**Comment by tonyroberts** [comment]
I've not really looked in too much detail I'm afraid, but I was thinking that if tp_bases on the type structure was set to a tuple of base classes, including the interfaces, then that would make isinstance work. From memory, I think currently tp_base is set and tp_bases is set to NULL.

**Comment by filmor** [comment]
Looking a bit deeper into this, I think this should be implemented in the metaclass instead, by means of [__instancecheck__](https://docs.python.org/2/reference/datamodel.html#customizing-instance-and-subclass-checks). If you add the interfaces to `tp_bases` it will fail on method resolution as `"mro() returned base class with unsuitable layout"`. In the end, the interfaces aren't really base classes anyway.

**Comment by denfromufa** [comment]
@filmor is this now implemented in your pull request here?https://github.com/pythonnet/pythonnet/pull/167

**Comment by filmor** [comment]
Yep, it is.

**Comment by tonyroberts** [comment]
Fixed in #167.
